### PR TITLE
CI: add Cirrus CI jobs

### PR DESCRIPTION
**Pull request by dnicolodi**
Run the tests on a few different Linux and Python distributions. I think this setup is much more agile than the Sage CI based one, and it catches issues much more effectively. The test are currently failing on issue I identified in one of my projects: the Debian jobs fail because of #109, fixed in https://github.com/pypa/pip/pull/11623 and the manylinux Python 3.7 job is failing because of https://github.com/mesonbuild/meson/issues/11097, fixed in https://github.com/mesonbuild/meson/pull/11104. 

Once the container images have been created and cached, the tests run in just a bit more than a minute and we can run 16 parallel configurations. I added a few relevant Linux distributions and Conda on Linux. Sage CI is also run on openSUSE but I don't know anything about package management on Suse thus I don't know how to create the container images. If someone contributes the image setup definition, it is trivial to add new jobs. I'm definitely not a container expert and I didn't do any effort to minimize the size of the images. Given that they are cached on the CI infrastructure, I don't think it matters.

Someone will need to figure out how to add the Cirrus CI GitHub Application to the repository and allow access to the logs.

### CODE REVIEW

**Comment by dnicolodi** [comment]
@jpakkane can you please install Cirrus CI for the meson-python repository? The instructions are here https://cirrus-ci.org/guide/quick-start/

**Comment by jpakkane** [comment]
Installed.

**Comment by FFY00** [comment]
Makes sense, and seems much faster. @mkoeppe do you have any thoughts?

**Comment by FFY00** [comment]
@dnicolodi can you look into the failure?

**Comment by dnicolodi** [comment]
The reason of the failure is detailed in the PR description 😃 "manylinux Python 3.7 job is failing because of https://github.com/mesonbuild/meson/issues/11097, fixed in https://github.com/mesonbuild/meson/pull/11104." I can find a way to have the job to use an unreleased Meson, but the Meson version fixing that should materialize soon so maybe we can just wait a bit.

**Comment by FFY00** [comment]
Ah sorry, my dyslexia sometimes makes it hard to parse stuff, and the ADHD messes with my concentration, I forgot you had already mentioned that :upside_down_face:

Yep, we can wait. But IMO, we could just merge it right now.

**Review by FFY00 — APPROVED** [review]
